tensorflow/core/kernels/non_max_suppression_op.cc

```diff
@@ -39,6 +39,17 @@ namespace {
 
 typedef Eigen::ThreadPoolDevice CPUDevice;
 
+// This function checks that all boxes have coherant ccordinates
+template <typename T>
+bool CheckBoxesCoordinates(const Tensor& boxes, int num_boxes){
+  typename TTypes<T, 2>::ConstTensor boxes_data = boxes.tensor<T, 2>();
+  for (int i = 0; i < num_boxes; ++i){
+    if(boxes_data(i, 0) == boxes_data(i, 2) || boxes_data(i, 1) == boxes_data(i, 3))
```

Comment:
This line still fails coding style: space after `if`, line length > 80 characters